x-pack/platform/plugins/shared/fleet/server/tasks/sync_integrations/sync_integrations_on_remote.ts

```diff
@@ -180,3 +215,219 @@ export const syncIntegrationsOnRemote = async (
+    const integrationsStatus = compareIntegrations(ccrIntegrations, installedIntegrationsByName);
+    const result = {
+      ...integrationsStatus,
+      ...(customAssetsStatus && { custom_assets: customAssetsStatus }),
+    };
+
+    return result;
+  } catch (err) {
+    logger.error('error', err.message);
+    throw err;
+  }
+};
+
+const compareIntegrations = (
+  ccrIntegrations: IntegrationsData[],
+  installedIntegrationsByName: Record<string, SavedObjectsFindResult<Installation>>
+) => {
+  const integrationsStatus: RemoteSyncedIntegrationsStatus[] | undefined = ccrIntegrations?.map(
+    (ccrIntegration) => {
+      const localIntegrationSO = installedIntegrationsByName[ccrIntegration.package_name];
+      if (!localIntegrationSO) {
+        return {
+          ...ccrIntegration,
+          sync_status: 'failed' as SyncStatus.FAILED,
+          error: `Installation not found`,
+        };
+      }
+      if (ccrIntegration.package_version !== localIntegrationSO?.attributes.version) {
+        return {
+          ...ccrIntegration,
+          sync_status: 'failed' as SyncStatus.FAILED,
+          error: `Installed version: ${localIntegrationSO?.attributes.version}`,
+        };
+      }
+      if (localIntegrationSO?.attributes.install_status !== 'installed') {
+        return {
+          ...ccrIntegration,
+          sync_status: 'failed' as SyncStatus.FAILED,
+          error: `Installation status: ${localIntegrationSO?.attributes.install_status}`,
+        };
+      }
+      return {
+        ...ccrIntegration,
+        sync_status: 'completed' as SyncStatus.COMPLETED,
+        updated_at: localIntegrationSO?.updated_at,
+      };
+    }
+  );
+  return { integrations: integrationsStatus };
+};
+
+const compareCustomAssets = async (
+  esClient: ElasticsearchClient,
+  logger: Logger,
+  ccrCustomAssets: { [key: string]: CustomAssetsData }
+) => {
+  if (!ccrCustomAssets) return;
+
+  const abortController = new AbortController();
+
+  try {
+    const installedPipelines = await getPipeline(esClient, abortController);
+    const installedComponentTemplates = await getComponentTemplate(esClient, abortController);
+
+    const componentTemplatesByName = (
+      installedComponentTemplates?.component_templates || []
+    ).reduce((acc, componentTemplate) => {
+      if (componentTemplate?.name) {
+        acc[componentTemplate.name] = componentTemplate.component_template?.template;
+      }
+      return acc;
+    }, {} as Record<string, ClusterComponentTemplateSummary>);
+    const result: RemoteSyncedCustomAssetsRecord = {};
+
+    // compare custom pipelines and custom component templates
+    Object.entries(ccrCustomAssets).forEach(([ccrCustomName, ccrCustomAsset]) => {
+      if (ccrCustomAsset.type === 'ingest_pipeline') {
+        const installedAsset = installedPipelines[ccrCustomAsset?.name];
+        if (isEqual(installedAsset?.processors, ccrCustomAsset?.pipeline?.processors)) {
```

Comment:
pipelines have an optional version which we can use to compare like here https://github.com/elastic/kibana/blob/b438a719ee34d0dfb19a7a6cbbafc616936581bb/x-pack/platform/plugins/shared/fleet/server/tasks/sync_integrations/custom_assets.ts#L241
we should probably have a common logic to compare here and `custom_assets.ts`